presto-test-jdbc-compatibility/src/test/java/io/prestosql/TestJdbcResultSetCompatibility.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql;
+
+import com.google.common.collect.ImmutableMap;
+import io.airlift.log.Logging;
+import io.prestosql.jdbc.TestJdbcResultSet;
+import io.prestosql.server.testing.TestingPrestoServer;
+import org.testng.SkipException;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.util.Optional;
+
+import static io.prestosql.JdbcDriverCapabilities.hasBrokenParametricTimestampWithTimeZoneSupport;
+import static io.prestosql.JdbcDriverCapabilities.testedVersion;
+import static io.prestosql.TestingServerUtils.setTestingServer;
+import static org.assertj.core.api.Assertions.assertThat;
+
+@Test(singleThreaded = true)
+public class TestJdbcResultSetCompatibility
+        extends TestJdbcResultSet
+{
+    private static final Optional<Integer> VERSION_UNDER_TEST = testedVersion();
+
+    @BeforeClass
+    @Override
+    public void setupServer()
+    {
+        Logging.initialize();
+        setTestingServer(this, TestingPrestoServer
+                .builder()
+                .setProperties(ImmutableMap.of("deprecated.omit-datetime-type-precision", "true"))
+                .build());
+    }
+
+    @AfterMethod(alwaysRun = true)
+    @Override
+    public void teardown()
+    {
+        // do not close connection and statement as this will fail
+    }
+
+    @Override
+    public void testObjectTypes()
+            throws Exception
+    {
+        if (hasBrokenParametricTimestampWithTimeZoneSupport()) {
+            throw new SkipException("This version raports PARAMETRIC_DATETIME client capability but TIMESTAMP WITH TIME ZONE is not supported");
+        }
+
+        super.testObjectTypes();
```

Comment:
It doesn't compile, since i split the method.
Since you need this for TIMESTAMP WITH TIME ZONE only, let's 
split the `testTimestamp` into `testTimestamp` and `testTimestampWithTimeZone`
(it was my error that i didn't split it), and override just  `testTimestampWithTimeZone`
here.

Reply:
I'm running this test for every old released version as well. So changing/splitting methods is not an option I guess.